### PR TITLE
Support basic auth in JSON RPC endpoint

### PR DESCRIPTION
**Pull request by vbaranov**
Resolves https://github.com/blockscout/blockscout/issues/3501

## Motivation

Blockscout doesn't support JSON RPC endpoints with basic auth.

## Changelog

Support basic auth in JSON RPC endpoints by adding hackney params: `[basic_auth: {"user", "pass"}]`.
`ETHEREUM_JSONRPC_USER ` and `ETHEREUM_JSONRPC_PASSWORD` are for basic auth management.

Docs update in https://github.com/blockscout/docs/pull/114.

## Checklist for your Pull Request (PR)

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [x] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [x] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [x] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

File: apps/explorer/config/dev/besu.exs [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+basic_auth_pass = System.get_env("ETHEREUM_JSONRPC_PASSWORD", "")
 
 hackney_opts =
-  if System.get_env("ETHEREUM_JSONRPC_HTTP_INSECURE", "") == "true" do
-    [:insecure] ++ hackney_opts_base
-  else
-    hackney_opts_base
-  end
+  [pool: :ethereum_jsonrpc]
+  |> (&if(System.get_env("ETHEREUM_JSONRPC_HTTP_INSECURE", "") == "true", do: [:insecure] ++ &1, else: &1)).()
```
Comment (nikitosing) [inline apps/explorer/config/dev/besu.exs:8]: Can we extract such boilerplate code to new separate module for example: `apps/explorer/config/helper.ex`(`Explorer.Config.Helper`). With one method:
```
defmodule Explorer.Config.Helper do
  def hackney_options() do
    basic_auth_user = System.get_env("ETHEREUM_JSONRPC_USER", "")
    basic_auth_pass = System.get_env("ETHEREUM_JSONRPC_PASSWORD", "")

    [pool: :ethereum_jsonrpc]
    |> (&if(System.get_env("ETHEREUM_JSONRPC_HTTP_INSECURE", "") == "true", do: [:insecure] ++ &1, else: &1)).()
    |> (&if(basic_auth_user != "" && basic_auth_pass != "",
          do: [basic_auth: {basic_auth_user, basic_auth_pass}] ++ &1,
          else: &1
        )).()
  end
end

```
And replace all 
```
basic_auth_user = System.get_env("ETHEREUM_JSONRPC_USER", "")
basic_auth_pass = System.get_env("ETHEREUM_JSONRPC_PASSWORD", "")

hackney_opts =
  [pool: :ethereum_jsonrpc]
  |> (&if(System.get_env("ETHEREUM_JSONRPC_HTTP_INSECURE", "") == "true", do: [:insecure] ++ &1, else: &1)).()
  |> (&if(basic_auth_user != "" && basic_auth_pass != "",
        do: [basic_auth: {basic_auth_user, basic_auth_pass}] ++ &1,
        else: &1
      )).()
```
on 
```
alias Explorer.Config.Helper

hackney_opts = Helper.hackney_options()
```

File: apps/explorer/config/dev/arbitrum.exs [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
 
 hackney_opts =
-  if System.get_env("ETHEREUM_JSONRPC_HTTP_INSECURE", "") == "true" do
-    [:insecure] ++ hackney_opts_base
-  else
-    hackney_opts_base
-  end
+  [pool: :ethereum_jsonrpc]
+  |> (&if(System.get_env("ETHEREUM_JSONRPC_HTTP_INSECURE", "") == "true", do: [:insecure] ++ &1, else: &1)).()
+  |> (&if(basic_auth_user != "" && basic_auth_pass != "",
```
Comment (nikitosing) [inline apps/explorer/config/dev/arbitrum.exs:9]: What if password is empty string?

Comment (vbaranov) [reply apps/explorer/config/dev/arbitrum.exs:9]: Does it make sense to enable Basic auth on RPC endpoint with blank password? I think it is very unlikely case.

Comment (nikitosing) [reply apps/explorer/config/dev/arbitrum.exs:9]: In some dev/test setups mb

Comment (vbaranov) [reply apps/explorer/config/dev/arbitrum.exs]: It is valid. Implemented in https://github.com/blockscout/blockscout/pull/6897/commits/3b5dc3db62700dff6cb5693e476613fa244edfb0.

**Review by nikitosing — APPROVED** [review]

**Pull request merged**